### DISCUSSION TITLE
Is it possible to show what is logged to `artifacts/logs` files also on stdout?

### DISCUSSION THREAD

**Discussion by xgqt** [dotnet/source-build#4082] [Q&A] [answered]
> Log: /var/notmp/portage/dev-dotnet/dotnet-sdk-8.0.101/work/dotnet-sdk-8.0.1/artifacts/logs/sdk.log

During build SDK will give a loty of info about environment but hide buid processes of individual components, is it possible to show contents of logs whenj the components are building?

This is important for Gentoo packaging as it helps us with bug reporting SDK build errors since we can have much more info in 1 logfile (the main one).

**Comment by omajid** [comment, marked as answer]
There are properties that can be passed to the build script to control the log output that's shown on the console.

- `MinimalConsoleLogOutput` can be set to  `false` to show component logs on the console
- `LogVerbosityOptOut` can be set to `false` to control what detail of information is logged
- `LogVerbosity` can be set to `quiet`/`minimal`/`normal`/`detailed` ([more here](https://learn.microsoft.com/en-us/visualstudio/msbuild/obtaining-build-logs-with-msbuild?view=vs-2022)) to control how much detail is displayed.

On Fedora, we use this build invocation:

```
VERBOSE=1 ./build.sh \
    --release-manifest %{SOURCE5} \
    -- \
    /p:MinimalConsoleLogOutput=false \
    /p:ContinueOnPrebuiltBaselineError=true \
    /v:n \
    /p:LogVerbosity=n \

```

**Comment by xgqt** [comment]
> VERBOSE=1 ./build.sh \
>     --release-manifest %{SOURCE5} \
>     -- \
>     /p:MinimalConsoleLogOutput=false \
>     /p:ContinueOnPrebuiltBaselineError=true \
>     /v:n \
>     /p:LogVerbosity=n \
> ```

Hello @omajid!

Thanks a lot for your explaination. Could you please tell what the other properties in the command you use are for? and why `VERBOSE=1`?

**Reply by omajid** [reply]
`--release-manifest` is documented in `--help`: https://github.com/dotnet/dotnet/blob/1bbd99e40c7f27a417fd310754f342f2a3dde761/build.sh#L12. You give it the `release.json` file if you are building offline outside of a git repo.

`ContinueOnPrebuiltBaselineError` turns off some error checking which, at least in my experience, doesn't really matter when building official releases and eliminates some noise on offline builds.

`VERBOSE=1` tells cmake (which .NET uses to compile C/C++ code) to show the individual compiler commands: https://stackoverflow.com/questions/2670121/using-cmake-with-gnu-make-how-can-i-see-the-exact-commands

**Reply by xgqt** [reply]
> `VERBOSE=1` tells cmake (which .NET uses to compile C/C++ code) to show the individual compiler commands: https://stackoverflow.com/questions/2670121/using-cmake-with-gnu-make-how-can-i-see-the-exact-commands

Ah, right, of course this affects cmake.

Thank you very much!
